### PR TITLE
MES-2205 Record serious fault against show me question

### PR DESCRIPTION
**Pull request by szabolcs-keresztes**
- Record serious fault against the show me question.
- Override tell me question driving fault badge with a serious fault badge after recording a serious fault on the show me question

IMG_6A47333AE8E7-1

### CODE REVIEW

File: src/pages/test-report/components/vehicle-check/vehicle-check.ts [hunk 1]
```diff
@@ -29,34 +43,93 @@ export class VehicleCheckComponent implements OnInit {
...
+    this.addOrRemoveFault();
   }
 
   onPress = () => {
-    console.log('competency pressed');
+    this.addOrRemoveFault(true);
   }
 
   canButtonRipple = () => {
+    // TODO: Implement this
```
Comment (rossfauldsbjss) [inline src/pages/test-report/components/vehicle-check/vehicle-check.ts]: Is this to be done in a different PR?

File: src/pages/test-report/components/vehicle-check/vehicle-check.ts [hunk 2]
```diff
@@ -29,34 +43,93 @@ export class VehicleCheckComponent implements OnInit {
...
+    if (this.isRemoveFaultMode) {
+      this.removeFault();
+      return;
+    }
+
+    this.addFault(wasPress);
+  }
+
+  removeFault = (): void => {
+    // TODO: Implement this
```
Comment (rossfauldsbjss) [inline src/pages/test-report/components/vehicle-check/vehicle-check.ts]: again - is this to be done in a separate PR?

File: src/pages/test-report/components/vehicle-check/vehicle-check.ts [hunk 3]
```diff
@@ -29,34 +43,93 @@ export class VehicleCheckComponent implements OnInit {
...
+
+    if (this.hasDangerousFault()) {
+      return 0;
+    }
+
+    if (this.tellMeQuestionFault !== CompetencyOutcome.DF) {
+      return 0;
+    }
+
+    // if show me question is not DF then return 0
```
Comment (rossfauldsbjss) [inline src/pages/test-report/components/vehicle-check/vehicle-check.ts]: comment placement is wrong, should be above the code it comments.

File: src/modules/tests/test_data/test-data.reducer.ts [hunk 4]
```diff
@@ -269,6 +272,7 @@ export function testDataReducer(
       return {
         ...state,
         vehicleChecks: {
+          ...state.vehicleChecks,
           tellMeQuestion: {
             ...state.vehicleChecks.tellMeQuestion,
             outcome: 'P',
```
Comment (buzzub) [inline src/modules/tests/test_data/test-data.reducer.ts]: Could this come from constants like `CompetencyOutcome` ?

File: src/pages/test-report/components/vehicle-check/vehicle-check.ts [hunk 5]
```diff
@@ -29,34 +43,93 @@ export class VehicleCheckComponent implements OnInit {
...
+      this.store$.dispatch(new ShowMeQuestionSeriousFault());
+      this.store$.dispatch(new ToggleSeriousFaultMode());
+    }
+
+    if (wasPress) {
+      // TODO: Implement this
+    }
+  }
+
   hasDrivingFault = (): number => {
```
Comment (buzzub) [inline src/pages/test-report/components/vehicle-check/vehicle-check.ts]: It feels like this should be called either `getFaultCount` and return a number or `hasDrivingFault` and return boolean.

File: src/pages/test-report/components/vehicle-check/vehicle-check.ts (hunk 3 above)
Comment (szabolcs-keresztes) [reply src/pages/test-report/components/vehicle-check/vehicle-check.ts]: It comments a code that does not yet exist. That's why it stands individually.

File: src/pages/test-report/components/vehicle-check/vehicle-check.ts (hunk 1 above)
Comment (szabolcs-keresztes) [reply src/pages/test-report/components/vehicle-check/vehicle-check.ts]: yes

File: src/pages/test-report/components/vehicle-check/vehicle-check.ts (hunk 2 above)
Comment (szabolcs-keresztes) [reply src/pages/test-report/components/vehicle-check/vehicle-check.ts]: yes

File: src/modules/tests/test_data/test-data.reducer.ts (hunk 4 above)
Comment (szabolcs-keresztes) [reply src/modules/tests/test_data/test-data.reducer.ts]: Yes, that would be good. We don't yet have it in the enum, but I can add it.

File: src/pages/test-report/components/vehicle-check/vehicle-check.ts (hunk 5 above)
Comment (szabolcs-keresztes) [reply src/pages/test-report/components/vehicle-check/vehicle-check.ts]: The driving-fault-badge requires a number to work. So I could rename the function to `getDrivingFaultCount`?

Comment (buzzub) [reply src/pages/test-report/components/vehicle-check/vehicle-check.ts]: I think I would

File: src/modules/tests/test_data/test-data.reducer.ts (hunk 4 above)
Comment (szabolcs-keresztes) [reply src/modules/tests/test_data/test-data.reducer.ts]: Added the `CompetencyOutcome.P`

File: src/pages/test-report/components/vehicle-check/vehicle-check.ts (hunk 5 above)
Comment (szabolcs-keresztes) [reply src/pages/test-report/components/vehicle-check/vehicle-check.ts]: Changed it
